File: media-visualizer-web/api/manwha-api.js
'use strict'

const manwhaApi = {
    url: `${apiBaseUrl}/Manwha`,

    get: async (id) => {
        const response = await fetch(`${manwhaApi.url}/${id}`);
        return response.json();
    },

    getList: async () => {
        const response = await fetch(`${manwhaApi.url}/GetList`);
        return response.json();
    },

    getRandom: async () => {
        const response = await fetch(`${manwhaApi.url}/GetRandom`);
        return response.json();
    }
};

File: MediaVisualizer.Web/Api/ManwhaApi.cs
using MediaVisualizer.Shared.Dtos;
using MediaVisualizer.Shared.Requests;
using MediaVisualizer.Shared.Responses;
using MediaVisualizer.Web.Helpers;

namespace MediaVisualizer.Web.Api;

public class ManwhaApi : IManwhaApi
{
    private const string BaseUrl = "Manwha";
    private readonly HttpClient _httpClient;

    public ManwhaApi(HttpClient httpClient)
    {
        _httpClient = httpClient;
    }

    public Task<ManwhaDto> GetAsync(int manwhaId)
    {
        return _httpClient.GetFromJsonAsync<ManwhaDto>($"{BaseUrl}/{manwhaId}");
    }

    public Task<ManwhaDto> GetRandomAsync()
    {
        return _httpClient.GetFromJsonAsync<ManwhaDto>($"{BaseUrl}/GetRandom");
    }

    public Task<List<string>> GetTitlesAsync()
    {
        return _httpClient.GetFromJsonAsync<List<string>>($"{BaseUrl}/GetTitles");
    }

    public Task<ListResponse<ManwhaDto>> GetListAsync(FiltersRequest filters)
    {
        var query = FiltersRequestHelper.BuildFiltersRequest(filters);
        return _httpClient.GetFromJsonAsync<ListResponse<ManwhaDto>>($"{BaseUrl}/GetList?{query}");
    }
}

public interface IManwhaApi
{
    Task<ManwhaDto> GetAsync(int manwhaId);
    Task<ListResponse<ManwhaDto>> GetListAsync(FiltersRequest filters);
    Task<ManwhaDto> GetRandomAsync();
    Task<List<string>> GetTitlesAsync();
}